languages/js/src/Query.ts

```diff
@@ -286,6 +288,8 @@ export class Query {
             for (const [k, v] of bindings) {
               transformed.set(k, await this.#host.toJs(v));
             }
+            rl?.emit('exit');
```

Comment:
This is kind of a mess but works. We need to "close" out our interaction with readline by removing the event handlers, so we fire the exit event which then adds back the handlers for the REPL. This call (`rl?.emit('exit')`) must be made any time this function yields, so could easily be messed up. Makes me think that debug should maybe be handled by a separate async generator that wraps this one.